File: frontend/src/types/ipRecord.ts
export type OSType = 'AIX' | 'Linux' | 'Windows';
export type IPStatus = 'Free' | 'Reserved' | 'In Use';
export type Environment = 'Production' | 'Test' | 'Development';

export interface IPRecord {
  id: string;
  ip_address: string;
  hostname: string | null;
  os_type: OSType;
  subnet_id: string;
  status: IPStatus;
  environment: Environment;
  owner: string | null;
  description: string | null;
  created_at: string;
  updated_at: string;
  created_by: string;
  updated_by: string;
  reserved_at: string | null;
  reserved_by: string | null;
}

export interface IPRecordCreate {
  ip_address: string;
  hostname?: string;
  os_type: OSType;
  subnet_id: string;
  status?: IPStatus;
  environment: Environment;
  owner?: string;
  description?: string;
}

export interface IPRecordUpdate {
  hostname?: string;
  os_type?: OSType;
  status?: IPStatus;
  environment?: Environment;
  owner?: string;
  description?: string;
}

export interface IPRecordFilters {
  subnet_id?: string;
  status?: IPStatus;
  os_type?: OSType;
  environment?: Environment;
  owner?: string;
  search?: string;
  page?: number;
  page_size?: number;
}

File: backend/app/models/ip_record.py
from enum import Enum
from typing import Optional
from datetime import datetime, timezone
from pydantic import BaseModel, Field, ConfigDict


class OSType(str, Enum):
    AIX = "AIX"
    LINUX = "Linux"
    WINDOWS = "Windows"


class IPStatus(str, Enum):
    FREE = "Free"
    RESERVED = "Reserved"
    IN_USE = "In Use"


class Environment(str, Enum):
    PRODUCTION = "Production"
    TEST = "Test"
    DEVELOPMENT = "Development"


class IPRecord(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: Optional[str] = Field(default=None, alias="_id")
    ip_address: str
    hostname: Optional[str] = None
    os_type: OSType
    subnet_id: str
    status: IPStatus = IPStatus.FREE
    environment: Environment
    owner: Optional[str] = None
    description: Optional[str] = None
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    created_by: str = "system"
    updated_by: str = "system"
    reserved_at: Optional[datetime] = None
    reserved_by: Optional[str] = None
